Build for Microchip XC32 compiler fails due to RCON name conflict

### Suggested enhancement

Static variable name change from **RCON** to something more descriptive.

### Justification

Mbed-TLS is used by mynewt project https://github.com/apache/mynewt-core.
It works fine with ARM and RISCV based compilers but is stopped to work with PIC32 XC32 compilers after some change in aes.c.

I guess eas.c includes assert.h which in turn for mynewt build also includes device specific headers and we have name conflict.

**aes.c** has static variable `RCON`

```c
static const uint32_t RCON[10] =
```
Unfortunately this directly conflicts with macro definition provided by PIC32 headers where RCON refers to some registers.

```txt
Error: In file included from c:\program files\microchip\xc32\v2.50\pic32mx\include\xc.h:689:0,
                 from repos/apache-mynewt-core/kernel/os/include/os/arch/pic32/os/os_arch.h:24,
                 from repos/apache-mynewt-core/kernel/os/include/os/os_fault.h:24,
                 from repos/apache-mynewt-core/libc/baselibc/include/assert.h:24,
                 from repos/mbedtls/library/common.h:32,
                 from repos/mbedtls/library/aes.c:26:
repos/mbedtls/library/aes.c:384:17: error: conflicting type qualifiers for 'RCON'
 static uint32_t RCON[10];
                 ^
c:\program files\microchip\xc32\v2.50\pic32mx\include\proc\pic32mz-ef\p32mz2048efh064.h:646:28: note: previous declaration of 'RCON' was here
 extern volatile uint32_t   RCON __attribute__((section("sfrs"), address(0xBF801240)));
```

I wander if Mbed-TLS project could change local static variable name to something else as this seems the easiest way to have crypto library work again for XC32 compiler.
This does not seem to impact library functionality but could make mynewt build work again.

Thanks

--- a/library/aes.c
+++ b/library/aes.c
@@ -360,7 +360,7 @@ static const uint32_t RT3[256] = { RT };
 /*
  * Round constants
  */
-static const uint32_t RCON[10] =
+static const uint32_t round_constants[10] =
 {
     0x00000001, 0x00000002, 0x00000004, 0x00000008,
     0x00000010, 0x00000020, 0x00000040, 0x00000080,
@@ -407,7 +407,7 @@ static uint32_t RT3[256];
 /*
  * Round constants
  */
-static uint32_t RCON[10];
+static uint32_t round_constants[10];
 
 /*
  * Tables generation code
@@ -438,7 +438,7 @@ static void aes_gen_tables(void)
      * calculate the round constants
      */
     for (i = 0, x = 1; i < 10; i++) {
-        RCON[i] = x;
+        round_constants[i] = x;
         x = XTIME(x);
     }
 
@@ -666,7 +666,7 @@ int mbedtls_aes_setkey_enc(mbedtls_aes_context *ctx, const unsigned char *key,
         case 10:
 
             for (unsigned int i = 0; i < 10; i++, RK += 4) {
-                RK[4]  = RK[0] ^ RCON[i] ^
+                RK[4]  = RK[0] ^ round_constants[i] ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_1(RK[3])]) ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_2(RK[3])] <<  8) ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_3(RK[3])] << 16) ^
@@ -682,7 +682,7 @@ int mbedtls_aes_setkey_enc(mbedtls_aes_context *ctx, const unsigned char *key,
         case 12:
 
             for (unsigned int i = 0; i < 8; i++, RK += 6) {
-                RK[6]  = RK[0] ^ RCON[i] ^
+                RK[6]  = RK[0] ^ round_constants[i] ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_1(RK[5])]) ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_2(RK[5])] <<  8) ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_3(RK[5])] << 16) ^
@@ -699,7 +699,7 @@ int mbedtls_aes_setkey_enc(mbedtls_aes_context *ctx, const unsigned char *key,
         case 14:
 
             for (unsigned int i = 0; i < 7; i++, RK += 8) {
-                RK[8]  = RK[0] ^ RCON[i] ^
+                RK[8]  = RK[0] ^ round_constants[i] ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_1(RK[7])]) ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_2(RK[7])] <<  8) ^
                          ((uint32_t) FSb[MBEDTLS_BYTE_3(RK[7])] << 16) ^

